### PR TITLE
Skip User.last_login_at saves up to 5 minutes after last write

### PR DESCRIPTION
**Pull request by nanotone**
### Description
This PR is intended to drastically reduce the number of writes to the `users` table.

Currently, every time [current_user](https://github.com/department-of-veterans-affairs/caseflow/blob/4d941af3bbfff26ae96468f0a87c66399e6c1e62/app/controllers/application_controller.rb#L66) is invoked by any code -- which is true for all but the most trivial HTTP endpoints -- the associated user record is updated with a new `last_login_at` value. By allowing `last_login_at` to be inaccurate up to 5 minutes, we can avoid making the vast majority of these database writes.

Credit for the idea goes to @yoomlam who suggested it during this [swat call](https://dsva.slack.com/archives/CHD7QU4L8/p1614610692016200) on reducing DB load.

### Testing Plan
1. New spec to test that `last_login_at` doesn't get updated more than once in a 5-minute interval.
2. New spec to test that `last_login_at` does get updated if more than 5 minutes have passed.

### CODE REVIEW

**Comment by alisan16** [comment]
Connects https://github.com/department-of-veterans-affairs/caseflow/issues/15565

**Comment by rajinderbhanot** [comment]
The column name last_login_at is a little misleading,  seems like the user's last action/activity (not just the login time).  The Board/HR does have future needs to track user activities for performance/usage reports etc.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -569,15 +569,13 @@
       it "updates last_login_at if it was more than 5 minutes ago" do
         user = create(:user, last_login_at: Time.zone.now - 10.minutes)
         session["user"]["pg_user_id"] = user.id
-        expect(subject).to eq user
-        expect(user.reload.last_login_at).to eq Time.zone.now
+        expect(subject.last_login_at).to eq Time.zone.now
```
Comment (yoomlam) [inline spec/models/user_spec.rb]: Hmmm... I actually don't mind the roundtrip, i.e., that it is comparing against the value stored in the database. I would have compared `to_i` values as suggested [here](https://stackoverflow.com/questions/20403063/trouble-comparing-time-with-rspec).
But this is acceptable.

**Review by yoomlam — APPROVED** [review]
Quick turnaround! 🎉

**Comment by yoomlam** [comment]
> The column name last_login_at is a little misleading, seems like the user's last action/activity (not just the login time). The Board/HR does have future needs to track user activities for performance/usage reports etc.

@rajinderbhanot Is a 5-minute precision sufficient for their needs?

**Comment by nanotone** [comment]
@rajinderbhanot You're correct, the sense of "login" here is referring not so much to the last time the user went through the eFolder/SSO/etc login flow, but rather the last time their user-agent (the browser) provided session credentials to the HTTP server.

I imagine that any future work to track user activity would be making use of more a suitable and business-aware mechanism than `last_login_at`. In any case, this is easy to undo, or tweak the desired imprecision.

**Pull request merged**